fsh/trialmatch_input.fsh

```diff
@@ -0,0 +1,39 @@
+Profile: TrialMatchInput
+Parent: Bundle
+Id: trial-match-input
+Title: "Trial Match Inputs"
+Description: "Trial Match Service Input Parameters"
+
+* type = #message
+// add all of the message dependent items such as message code, the messages header as the first item in the entry array
+
+* entry ^slicing.discriminator.type = #profile
+* entry ^slicing.discriminator.path = "entry.resource.resolve()"
+//SUSHI error this does not exist * entry ^slicing.discriminator.strategy = includes
+* entry ^slicing.rules = #open
+* entry contains
+    matchMessageHeader 0..1 and
```

Comment:
Will need to check and see if slicing enforces order here or not.  Messageheader must be the first entry and may have to be defined explicitly in that position via entry[0].resource only MesageHeader or something along those lines.  Same may need to apply to Parameters as well.  

We also probably do not want to force order of the rest of the items in the bundle and may also need to make sure that the slicing rules are closed so no other resources can be provided.